### DISCUSSION TITLE
Comments

### DISCUSSION THREAD

**Discussion by EliteMasterEric** [EliteMasterEric/Teyvat.moe#252] [Ideas]
I had a wonderful, awful idea.

- [ ] Wait for GitHub Discussions to leave beta.
- [ ] Add https://www.npmjs.com/package/@octokit/rest
- [ ] Add a feature for the user to authenticate with GitHub via OAuth.
- [ ] Add a panel which displays comments on markers.
  - [ ] Search this GitHub repository for a discussion whose name includes the ID of the marker. If one doesn't exist, display no comments.
  - [ ] If a discussion does exist, display its comments in order by votes.
  - [ ] Users can type a comment, and clicking submit will create an API call to comment on a discussion (creating one if it doesn't exist).
  - [ ] Users can click to vote on comments.

The discussions page will turn into a cluttered mess unless discussions could be made to be unlisted or hidden, but this would also act as a database to access user feedback (with voting!) on markers.

**Comment by Alkmaaar** [comment]
So.. is it like genshin map appsamle, i suppose. This is really good idea!

**Reply by EliteMasterEric** [reply]
This feature would create a topic in this area of GitHub for each marker that has discussions, and the map would check if there's a discussion here for a given marker.

The end result should look like AppSample. The main issue is that the Discussion feature is in beta, and the necessary features for implementing this (namely an API so I can access the data, and the voting system which is [only active on select repositories](https://github.com/github/feedback/discussions)) are not available yet, and we have to wait on GitHub for those.
